csrc/scheduler/reduction_utils.cpp

```diff
@@ -732,8 +733,18 @@ class PersistentBufferProjector {
                 buffer) == projectable_persistent_buffers.end()) {
           continue;
         }
+        std::vector<Val*> vals_project_to = fusion_->inputs();
```

Comment:
Please add some more comments here. What the issue is, how it is addressed, any potential concern? An example would be also very helpful.

Reply:
Also, please be more explicit what are assumed. One important assumption here is that since this is just for normalizations, a reduction must be always followed by a broadcast.

Reply:
Also, if there's a path from a reduction output to the persistent buffer without having a broadcast, we would end up recomputing the reduction. I suspect this  pattern would still be a valid persistent fusion, although it should be uncommon in practice. I think we should check if all the paths between a reduction output to a persistent buffer contains a broadcast.